### ISSUE TITLE
"robots.txt" not working

### ISSUE THREAD

**Issue by lcgkm** [hashicorp/consul#5005]
Please serve robots.txt from the root when the UI is enabled. 
Currently if the UI is enabled, the robots.txt file is in the wrong place(/ui/robots.txt). It is not recommended to run Consul exposed to the internet, but in the event that it happens, this will prevent Consul's UI page from being crawled by search engine spiders.

`
coupang@ip-10-211-178-16:~$ consul version 
Consul v1.4.0+ent 
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
`
Reference: 
https://github.com/hashicorp/vault/issues/5585

**Comment by remilapeyre** [comment]
Serving `robots.txt` from a subfolder is indeed useless.

The fix could be as simple as
```
diff --git a/agent/http.go b/agent/http.go
index 73e9925a2..40344d307 100644
--- a/agent/http.go
+++ b/agent/http.go
@@ -219,6 +219,7 @@ func (s *HTTPServer) handler(enableDebug bool) http.Handler {
                        uifs = &redirectFS{fs: uifs}
                }

+               mux.Handle("/robots.txt", http.FileServer(uifs))
                mux.Handle("/ui/", http.StripPrefix("/ui/", http.FileServer(uifs)))
        }
```

but this would leave `/ui/robots.txt` as well (though this is not really an issue).

**Comment by ChipV223** [comment]
@lcgkm : The fix was this was merged with #5089, so you can try to build a new version from master to test it
